typing/includemod.ml

```diff
@@ -736,6 +758,7 @@ and signature_components  ~in_eq ~loc old_env ~mark env subst
             let shape_map =
               Shape.Map.add_module_type_proj shape_map id1 orig_shape
             in
+            let item = unrecoverable_error item in
```

Comment:
This is fishy. We can match `val x : t1` against `val x : t2`, recover and continue, but we cannot recover from `module type S : sig val x : t1 end` against `module type S : sig val x : t2 end`?

Reply:
The current fix makes no effort to identify all recoverable errors, since recoverable errors will only be useful in a future error message improvements.